torch/utils/data/datapipes/_typing.py

```diff
@@ -375,6 +402,15 @@ def wrap_generator(*args, **kwargs):
                         response = gen.send(request)
             except StopIteration as e:
                 return e.value
+            except Exception as e:
+                # TODO: Simplify the traceback message to skip over `response = gen.send(None)`
+                #       Part of https://github.com/pytorch/data/issues/284
+                datapipe = args[0]
+                msg = "thrown by __iter__ of"
+                full_msg = f"{msg} {datapipe.__class__.__name__}({_generate_input_args_string(datapipe)})"
+                if len(e.args) >= 1 and msg not in e.args[0]:
+                    e.args = (e.args[0] + f'\n This exception is {full_msg}',) + e.args[1:]
```

Comment:
I have one concern on this approach. Seeing from python document (https://docs.python.org/3/library/exceptions.html#BaseException.args), does this work for `OSError`?

Reply:
It seems working. NVM.